### PR TITLE
resolver: manual resolver crashes if grpc.Dial isn't called before some methods

### PR DESCRIPTION
**Pull request by henvic**
Add panic message when manual resolver crashes if `grpc.Dial()` isn't called before setting ClientConn

RELEASE NOTES: none

### CODE REVIEW

File: resolver/manual/manual.go [hunk 1]
```diff
@@ -114,6 +119,10 @@ func (r *Resolver) UpdateState(s resolver.State) {
 // ReportError calls CC.ReportError.
 func (r *Resolver) ReportError(err error) {
 	r.mu.Lock()
+	if r.CC == nil {
+		r.mu.Unlock()
```
Comment (dfawley) [inline resolver/manual/manual.go:123]: As above re: `defer` except there is no concern about callbacks in this case.

File: resolver/manual/manual_test.go [hunk 2]
```diff
@@ -1,3 +1,21 @@
+/*
+ *
+ * Copyright 2017 gRPC authors.
```
Comment (dfawley) [inline resolver/manual/manual_test.go:3]: Please update to 2023.

**Review by dfawley — APPROVED** [review]
LGTM modulo the date thing.  Thanks!

**Comment by dfawley** [comment]
@arvindbr8 PTAL; thanks!

File: resolver/manual/manual_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+func TestResolver(t *testing.T) {
+	r := manual.NewBuilderWithScheme("whatever")
+	r.InitialState(resolver.State{
+		Addresses: []resolver.Address{
+			{Addr: "address"},
+		},
+	})
+
+	t.Run("update_state", func(t *testing.T) {
```
Comment (arvindbr8) [inline resolver/manual/manual_test.go:39]: I would prefer if the sub test description said not just `update_state`. This subtest asserts the panic message when updateState fails. So maybe something like 

```suggestion
	t.Run("update_state_panics", func(t *testing.T) {
```

File: resolver/manual/manual_test.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+			if r := recover(); r != want {
+				t.Errorf("expected panic %q, got %q", want, r)
+			}
+		}()
+		r.UpdateState(resolver.State{Addresses: []resolver.Address{
+			{Addr: "address"},
+			{Addr: "anotheraddress"},
+		}})
+	})
+	t.Run("report_error", func(t *testing.T) {
```
Comment (arvindbr8) [inline resolver/manual/manual_test.go:51]: Similarly for `report_error`

File: resolver/manual/manual_test.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+		defer func() {
+			want := "cannot report error as grpc.Dial with resolver has not been called"
+			if r := recover(); r != want {
+				t.Errorf("expected panic %q, got %q", want, r)
+			}
+		}()
+		r.ReportError(errors.New("example"))
+	})
+
+	_, err := grpc.Dial("whatever://localhost",
```
Comment (arvindbr8) [inline resolver/manual/manual_test.go:61]: This seems like another subTest to me. I would prefer if we did this also inside of a `t.Run("happy path", func(t *testing.T)`

**Review by arvindbr8 — CHANGES_REQUESTED** [review]

**Review by arvindbr8 — APPROVED** [review]

**Pull request merged**